Cannot load Knative configuration from file or env variable

### What happened?

Installing pure Knative Eventing (just upstream knative eventing) and Camel-K 2.3 via helm, from scripts:

* https://github.com/matzew/eventing-cloud-integraton/blob/main/system/01-camel-k.sh
* https://github.com/matzew/eventing-cloud-integraton/blob/main/system/02-kn-eventing.sh

But unable to use Eventing with a simple "beer source" kamelet/pipe. Doing so gives me the following in the pod of the pipe:

```
2024-04-23 09:28:42,806 INFO  [org.apa.cam.mai.MainSupport] (main) Apache Camel (Main) 4.4.1 is starting
2024-04-23 09:28:42,835 INFO  [org.apa.cam.k.sup.SourcesSupport] (main) Loading routes from: SourceDefinition{name='camel-k-embedded-flow', language='yaml', type='source', location='file:/etc/camel/sources/camel-k-embedded-flow.yaml', }
2024-04-23 09:28:42,928 ERROR [org.apa.cam.qua.mai.CamelMainRuntime] (main) Failed to start application: org.apache.camel.FailedToCreateRouteException: Failed to create route binding at: >>> To[knative:event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet] <<< in route: Route(binding)[From[kamelet:beer-source/source] -> [To[knati... because of Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
...
...
```

In the past I had to add a few hacks to get this working, but w/ the new 2.3.0 release out there I removed those hacks yesterday:
https://github.com/matzew/eventing-cloud-integraton/commit/c607fa88eb065fc2b40193b7fa5ead1dcfe74ebb

The fact that the error above says **Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker** seem to me wrong, b/c the addressable broker is there. and is ready:

```
k get brokers.eventing.knative.dev -A
NAMESPACE   NAME      URL                                                                        AGE   READY   REASON
default     kamelet   http://broker-ingress.knative-eventing.svc.cluster.local/default/kamelet   15m   True    
```

### Steps to reproduce

1. run the install from the above repo
2. make use of the `samples` app folder on the repo (https://github.com/matzew/eventing-cloud-integraton/tree/main/samples/01-local-integrations)
3. log the pod
4. see the error

### Relevant log output

```shell
exec java -cp ./resources:/etc/camel/application.properties:/etc/camel/conf.d/_resources:/etc/camel/kamelets/kamelets-bundle-beer-source-pipe-001:/etc/camel/resources:/etc/camel/sources/camel-k-embedded-flow.yaml:dependencies/app/camel-k-integration-2.3.0.jar:dependencies/lib/boot/io.github.crac.org-crac-0.1.3.jar:dependencies/lib/boot/io.quarkus.quarkus-bootstrap-runner-3.8.3.jar:dependencies/lib/boot/io.quarkus.quarkus-development-mode-spi-3.8.3.jar:dependencies/lib/boot/io.quarkus.quarkus-vertx-latebound-mdc-provider-3.8.3.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-constraint-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-cpu-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-expression-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-function-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-io-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-net-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-os-2.1.2.jar:dependencies/lib/boot/io.smallrye.common.smallrye-common-ref-2.1.2.jar:dependencies/lib/boot/jakarta.json.jakarta.json-api-2.1.3.jar:dependencies/lib/boot/org.eclipse.parsson.parsson-1.1.5.jar:dependencies/lib/boot/org.jboss.logging.jboss-logging-3.5.3.Final.jar:dependencies/lib/boot/org.jboss.logmanager.jboss-logmanager-3.0.4.Final.jar:dependencies/lib/main/com.aayushatharva.brotli4j.brotli4j-1.16.0.jar:dependencies/lib/main/com.aayushatharva.brotli4j.native-linux-x86_64-1.16.0.jar:dependencies/lib/main/com.aayushatharva.brotli4j.service-1.16.0.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-annotations-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-core-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.core.jackson-databind-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.dataformat.jackson-dataformat-yaml-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.datatype.jackson-datatype-jdk8-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.16.1.jar:dependencies/lib/main/com.fasterxml.jackson.module.jackson-module-parameter-names-2.16.1.jar:dependencies/lib/main/com.github.mifmif.generex-1.0.2.jar:dependencies/lib/main/commons-codec.commons-codec-1.16.1.jar:dependencies/lib/main/commons-io.commons-io-2.15.1.jar:dependencies/lib/main/dk.brics.automaton.automaton-1.11-8.jar:dependencies/lib/main/io.fabric8.kubernetes-client-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-client-api-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-httpclient-vertx-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-admissionregistration-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-apiextensions-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-apps-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-autoscaling-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-batch-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-certificates-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-common-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-coordination-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-core-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-discovery-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-events-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-extensions-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-flowcontrol-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-gatewayapi-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-metrics-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-networking-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-node-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-policy-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-rbac-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-resource-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-scheduling-6.10.0.jar:dependencies/lib/main/io.fabric8.kubernetes-model-storageclass-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-client-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-client-api-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-clusterautoscaling-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-config-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-console-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-installer-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-machine-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-machineconfig-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-monitoring-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-operatorhub-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-storageversionmigrator-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-tuned-6.10.0.jar:dependencies/lib/main/io.fabric8.openshift-model-whereabouts-6.10.0.jar:dependencies/lib/main/io.fabric8.zjsonpatch-0.3.0.jar:dependencies/lib/main/io.netty.netty-buffer-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-dns-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-haproxy-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-http-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-http2-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-codec-socks-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-common-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-handler-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-handler-proxy-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-resolver-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-resolver-dns-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-transport-4.1.107.Final.jar:dependencies/lib/main/io.netty.netty-transport-native-unix-common-4.1.107.Final.jar:dependencies/lib/main/io.quarkus.arc.arc-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-apache-httpclient-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-arc-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-core-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-credentials-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-fs-util-0.0.10.jar:dependencies/lib/main/io.quarkus.quarkus-jackson-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-jsonp-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-kubernetes-client-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-kubernetes-client-internal-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-logging-json-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-mutiny-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-netty-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-security-runtime-spi-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-smallrye-context-propagation-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-vertx-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-vertx-http-3.8.3.jar:dependencies/lib/main/io.quarkus.quarkus-virtual-threads-3.8.3.jar:dependencies/lib/main/io.quarkus.security.quarkus-security-2.0.3.Final.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-annotation-2.1.2.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-classloader-2.1.2.jar:dependencies/lib/main/io.smallrye.common.smallrye-common-vertx-context-2.1.2.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-3.5.4.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-common-3.5.4.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-core-3.5.4.jar:dependencies/lib/main/io.smallrye.config.smallrye-config-source-yaml-3.5.4.jar:dependencies/lib/main/io.smallrye.reactive.mutiny-2.5.8.jar:dependencies/lib/main/io.smallrye.reactive.mutiny-smallrye-context-propagation-2.5.8.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-auth-common-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-bridge-common-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-core-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-runtime-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-uri-template-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-web-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-web-common-3.10.0.jar:dependencies/lib/main/io.smallrye.reactive.vertx-mutiny-generator-3.10.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-2.1.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-api-2.1.0.jar:dependencies/lib/main/io.smallrye.smallrye-context-propagation-storage-2.1.0.jar:dependencies/lib/main/io.smallrye.smallrye-fault-tolerance-vertx-6.2.6.jar:dependencies/lib/main/io.vertx.vertx-auth-common-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-bridge-common-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-codegen-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-core-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-uri-template-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-web-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-web-client-4.5.4.jar:dependencies/lib/main/io.vertx.vertx-web-common-4.5.4.jar:dependencies/lib/main/jakarta.activation.jakarta.activation-api-2.1.2.jar:dependencies/lib/main/jakarta.annotation.jakarta.annotation-api-2.1.1.jar:dependencies/lib/main/jakarta.el.jakarta.el-api-5.0.1.jar:dependencies/lib/main/jakarta.enterprise.jakarta.enterprise.cdi-api-4.0.1.jar:dependencies/lib/main/jakarta.enterprise.jakarta.enterprise.lang-model-4.0.1.jar:dependencies/lib/main/jakarta.inject.jakarta.inject-api-2.0.1.jar:dependencies/lib/main/jakarta.interceptor.jakarta.interceptor-api-2.1.0.jar:dependencies/lib/main/jakarta.servlet.jakarta.servlet-api-6.0.0.jar:dependencies/lib/main/jakarta.transaction.jakarta.transaction-api-2.0.1.jar:dependencies/lib/main/jakarta.xml.bind.jakarta.xml.bind-api-4.0.1.jar:dependencies/lib/main/org.apache.camel.camel-api-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-attachments-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-base-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-base-engine-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-bean-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-cloud-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-cloudevents-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-cluster-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-componentdsl-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-catalog-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-engine-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-languages-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-model-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-processor-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-core-reifier-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-dsl-support-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-endpointdsl-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-file-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-health-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-http-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-http-base-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-http-common-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-kamelet-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-knative-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-knative-api-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-knative-http-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-kubernetes-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-main-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-management-api-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-microprofile-config-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-support-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-timer-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-tooling-model-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-util-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-util-json-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-xml-jaxp-util-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-yaml-dsl-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-yaml-dsl-common-4.4.1.jar:dependencies/lib/main/org.apache.camel.camel-yaml-dsl-deserializers-4.4.1.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-3.8.1.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-api-3.8.1.jar:dependencies/lib/main/org.apache.camel.k.camel-k-core-support-3.8.1.jar:dependencies/lib/main/org.apache.camel.k.camel-k-runtime-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-bean-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-cloudevents-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-core-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-core-cloud-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-http-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-http-common-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-kamelet-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-knative-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-kubernetes-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-commons-logging-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-support-httpclient-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-timer-3.8.1.jar:dependencies/lib/main/org.apache.camel.quarkus.camel-quarkus-yaml-dsl-3.8.1.jar:dependencies/lib/main/org.apache.commons.commons-compress-1.26.1.jar:dependencies/lib/main/org.apache.commons.commons-lang3-3.14.0.jar:dependencies/lib/main/org.apache.httpcomponents.client5.httpclient5-5.2.1.jar:dependencies/lib/main/org.apache.httpcomponents.core5.httpcore5-5.2.1.jar:dependencies/lib/main/org.apache.httpcomponents.core5.httpcore5-h2-5.2.4.jar:dependencies/lib/main/org.apache.httpcomponents.httpclient-4.5.14.jar:dependencies/lib/main/org.apache.httpcomponents.httpcore-4.4.16.jar:dependencies/lib/main/org.brotli.dec-0.1.2.jar:dependencies/lib/main/org.eclipse.angus.angus-activation-2.0.1.jar:dependencies/lib/main/org.eclipse.microprofile.config.microprofile-config-api-3.0.3.jar:dependencies/lib/main/org.eclipse.microprofile.context-propagation.microprofile-context-propagation-api-1.3.jar:dependencies/lib/main/org.jboss.logging.commons-logging-jboss-logging-1.0.0.Final.jar:dependencies/lib/main/org.jboss.logging.jboss-logging-annotations-2.2.1.Final.jar:dependencies/lib/main/org.jboss.slf4j.slf4j-jboss-logmanager-2.0.0.Final.jar:dependencies/lib/main/org.jboss.threads.jboss-threads-3.5.1.Final.jar:dependencies/lib/main/org.slf4j.slf4j-api-2.0.6.jar:dependencies/lib/main/org.snakeyaml.snakeyaml-engine-2.7.jar:dependencies/lib/main/org.wildfly.common.wildfly-common-1.7.0.Final.jar:dependencies/lib/main/org.yaml.snakeyaml-2.2.jar:dependencies/quarkus-app-dependencies.txt:dependencies/quarkus-run.jar:dependencies/quarkus/generated-bytecode.jar:dependencies/quarkus/quarkus-application.dat:dependencies/quarkus/transformed-bytecode.jar io.quarkus.bootstrap.runner.QuarkusEntryPoint
2024-04-23 09:28:42,800 INFO  [org.apa.cam.k.Runtime] (main) Apache Camel K Runtime 3.8.1
2024-04-23 09:28:42,804 INFO  [org.apa.cam.qua.cor.CamelBootstrapRecorder] (main) Bootstrap runtime: org.apache.camel.quarkus.main.CamelMainRuntime
2024-04-23 09:28:42,806 INFO  [org.apa.cam.mai.MainSupport] (main) Apache Camel (Main) 4.4.1 is starting
2024-04-23 09:28:42,835 INFO  [org.apa.cam.k.sup.SourcesSupport] (main) Loading routes from: SourceDefinition{name='camel-k-embedded-flow', language='yaml', type='source', location='file:/etc/camel/sources/camel-k-embedded-flow.yaml', }
2024-04-23 09:28:42,928 ERROR [org.apa.cam.qua.mai.CamelMainRuntime] (main) Failed to start application: org.apache.camel.FailedToCreateRouteException: Failed to create route binding at: >>> To[knative:event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet] <<< in route: Route(binding)[From[kamelet:beer-source/source] -> [To[knati... because of Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:228)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:79)
	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:728)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:594)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2385)
	at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:194)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:78)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2021)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:105)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2040)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:212)
	at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:113)
	at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:140)
	at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot173480958.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot173480958.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:830)
	at org.apache.camel.impl.engine.AbstractCamelContext.getEndpoint(AbstractCamelContext.java:730)
	at org.apache.camel.support.CamelContextHelper.resolveEndpoint(CamelContextHelper.java:123)
	at org.apache.camel.reifier.SendReifier.resolveEndpoint(SendReifier.java:47)
	at org.apache.camel.reifier.SendReifier.createProcessor(SendReifier.java:37)
	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:879)
	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:619)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:224)
	... 31 more
Caused by: java.lang.IllegalStateException: Cannot load Knative configuration from file or env variable
	at org.apache.camel.component.knative.KnativeComponent.getKnativeConfiguration(KnativeComponent.java:288)
	at org.apache.camel.component.knative.KnativeComponent.createEndpoint(KnativeComponent.java:226)
	at org.apache.camel.support.DefaultComponent.createEndpoint(DefaultComponent.java:170)
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:796)
	... 38 more

2024-04-23 09:28:42,949 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: org.apache.camel.FailedToCreateRouteException: Failed to create route binding at: >>> To[knative:event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet] <<< in route: Route(binding)[From[kamelet:beer-source/source] -> [To[knati... because of Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:47)
	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot173480958.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot173480958.deploy(Unknown Source)
	... 13 more
Caused by: java.lang.RuntimeException: org.apache.camel.FailedToCreateRouteException: Failed to create route binding at: >>> To[knative:event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet] <<< in route: Route(binding)[From[kamelet:beer-source/source] -> [To[knati... because of Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:65)
	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
	... 15 more
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route binding at: >>> To[knative:event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet] <<< in route: Route(binding)[From[kamelet:beer-source/source] -> [To[knati... because of Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:228)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:79)
	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:728)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:594)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2385)
	at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:194)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:78)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2021)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:105)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2040)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:212)
	at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:113)
	at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:140)
	at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
	... 16 more
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: knative://event?apiVersion=eventing.knative.dev%2Fv1&cloudEventsType=com.corp.my.beer.source&kind=Broker&name=kamelet due to: Cannot load Knative configuration from file or env variable
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:830)
	at org.apache.camel.impl.engine.AbstractCamelContext.getEndpoint(AbstractCamelContext.java:730)
	at org.apache.camel.support.CamelContextHelper.resolveEndpoint(CamelContextHelper.java:123)
	at org.apache.camel.reifier.SendReifier.resolveEndpoint(SendReifier.java:47)
	at org.apache.camel.reifier.SendReifier.createProcessor(SendReifier.java:37)
	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:879)
	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:619)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:224)
	... 31 more
Caused by: java.lang.IllegalStateException: Cannot load Knative configuration from file or env variable
	at org.apache.camel.component.knative.KnativeComponent.getKnativeConfiguration(KnativeComponent.java:288)
	at org.apache.camel.component.knative.KnativeComponent.createEndpoint(KnativeComponent.java:226)
	at org.apache.camel.support.DefaultComponent.createEndpoint(DefaultComponent.java:170)
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:796)
```


### Camel K version

2.3.0

--- a/pkg/controller/integration/platform_setup.go
+++ b/pkg/controller/integration/platform_setup.go
@@ -97,7 +97,7 @@ func determineBestTraitProfile(c client.Client, integration *v1.Integration, p *
 		// Use platform spec profile if set
 		return p.Spec.Profile, nil
 	}
-	if ok, err := knative.IsServingInstalled(c); err != nil {
+	if ok, err := knative.IsInstalled(c); err != nil {
 		return "", err
 	} else if ok {
 		return v1.TraitProfileKnative, nil

--- a/pkg/controller/kameletbinding/integration.go
+++ b/pkg/controller/kameletbinding/integration.go
@@ -248,7 +248,7 @@ func determineTraitProfile(ctx context.Context, c client.Client, binding *v1alph
 			return pl.Spec.Profile, nil
 		}
 	}
-	if ok, err := knative.IsServingInstalled(c); err != nil {
+	if ok, err := knative.IsInstalled(c); err != nil {
 		return "", err
 	} else if ok {
 		return v1.TraitProfileKnative, nil

--- a/pkg/controller/pipe/integration.go
+++ b/pkg/controller/pipe/integration.go
@@ -248,7 +248,7 @@ func determineTraitProfile(ctx context.Context, c client.Client, binding *v1.Pip
 			return pl.Spec.Profile, nil
 		}
 	}
-	if ok, err := knative.IsServingInstalled(c); err != nil {
+	if ok, err := knative.IsInstalled(c); err != nil {
 		return "", err
 	} else if ok {
 		return v1.TraitProfileKnative, nil

--- a/pkg/trait/gc.go
+++ b/pkg/trait/gc.go
@@ -47,6 +47,7 @@ import (
 	v1 "github.com/apache/camel-k/v2/pkg/apis/camel/v1"
 	traitv1 "github.com/apache/camel-k/v2/pkg/apis/camel/v1/trait"
 	"github.com/apache/camel-k/v2/pkg/util"
+	"github.com/apache/camel-k/v2/pkg/util/knative"
 	"github.com/apache/camel-k/v2/pkg/util/log"
 )
 
@@ -85,20 +86,6 @@ var (
 			Group:   batchv1.SchemeGroupVersion.Group,
 			Version: batchv1.SchemeGroupVersion.Version,
 		}: {},
-	}
-	deletableTypesByProfile = map[v1.TraitProfile]map[schema.GroupVersionKind]struct{}{
-		v1.TraitProfileKnative: {
-			schema.GroupVersionKind{
-				Kind:    "Service",
-				Group:   "serving.knative.dev",
-				Version: "v1",
-			}: {},
-			schema.GroupVersionKind{
-				Kind:    "Trigger",
-				Group:   "eventing.knative.dev",
-				Version: "v1",
-			}: {},
-		},
 	}
 )
 
@@ -160,12 +147,30 @@ func (t *gcTrait) garbageCollectResources(e *Environment) error {
 	}
 
 	profile := e.DetermineProfile()
-	if profileDeletableTypes, ok := deletableTypesByProfile[profile]; ok {
-		// copy profile related deletable types if not already present
-		for key, value := range profileDeletableTypes {
-			if _, found := deletableGVKs[key]; !found {
-				deletableGVKs[key] = value
-			}
+	deletableTypesByProfile := map[schema.GroupVersionKind]struct{}{}
+
+	if profile == v1.TraitProfileKnative {
+		if ok, _ := knative.IsServingInstalled(e.Client); ok {
+			deletableTypesByProfile[schema.GroupVersionKind{
+				Kind:    "Service",
+				Group:   "serving.knative.dev",
+				Version: "v1",
+			}] = struct{}{}
+		}
+
+		if ok, _ := knative.IsEventingInstalled(e.Client); ok {
+			deletableTypesByProfile[schema.GroupVersionKind{
+				Kind:    "Trigger",
+				Group:   "eventing.knative.dev",
+				Version: "v1",
+			}] = struct{}{}
+		}
+	}
+
+	// copy profile related deletable types if not already present
+	for key, value := range deletableTypesByProfile {
+		if _, found := deletableGVKs[key]; !found {
+			deletableGVKs[key] = value
 		}
 	}
 

--- a/pkg/trait/knative.go
+++ b/pkg/trait/knative.go
@@ -331,7 +331,9 @@ func (t *knativeTrait) configureEvents(e *Environment, env *knativeapi.CamelEnvi
 				serviceName = "default"
 			}
 			servicePath := fmt.Sprintf("/events/%s", eventType)
-			t.createTrigger(e, ref, eventType, servicePath)
+			if triggerErr := t.createTrigger(e, ref, eventType, servicePath); triggerErr != nil {
+				return triggerErr
+			}
 
 			if !env.ContainsService(serviceName, knativeapi.CamelEndpointKindSource, knativeapi.CamelServiceTypeEvent, ref.APIVersion, ref.Kind) {
 				svc := knativeapi.CamelServiceDefinition{
@@ -475,7 +477,7 @@ func (t *knativeTrait) configureSinkBinding(e *Environment, env *knativeapi.Came
 	return err
 }
 
-func (t *knativeTrait) createTrigger(e *Environment, ref *corev1.ObjectReference, eventType string, path string) {
+func (t *knativeTrait) createTrigger(e *Environment, ref *corev1.ObjectReference, eventType string, path string) error {
 	// TODO extend to additional filters too, to filter them at source and not at destination
 	found := e.Resources.HasKnativeTrigger(func(trigger *eventing.Trigger) bool {
 		return trigger.Spec.Broker == ref.Name &&
@@ -486,9 +488,32 @@ func (t *knativeTrait) createTrigger(e *Environment, ref *corev1.ObjectReference
 		if ref.Namespace == "" {
 			ref.Namespace = e.Integration.Namespace
 		}
-		trigger := knativeutil.CreateTrigger(*ref, e.Integration.Name, eventType, path)
+
+		controllerStrategy, err := e.DetermineControllerStrategy()
+		if err != nil {
+			return err
+		}
+
+		var trigger *eventing.Trigger
+		switch controllerStrategy {
+		case ControllerStrategyKnativeService:
+			trigger, err = knativeutil.CreateKnativeServiceTrigger(*ref, e.Integration.Name, eventType, path)
+			if err != nil {
+				return err
+			}
+		case ControllerStrategyDeployment:
+			trigger, err = knativeutil.CreateServiceTrigger(*ref, e.Integration.Name, eventType, path)
+			if err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("failed to create Knative trigger: unsupported controller strategy %s", controllerStrategy)
+		}
+
 		e.Resources.Add(trigger)
 	}
+
+	return nil
 }
 
 func (t *knativeTrait) ifServiceMissingDo(

--- a/pkg/trait/knative_service.go
+++ b/pkg/trait/knative_service.go
@@ -30,6 +30,7 @@ import (
 	v1 "github.com/apache/camel-k/v2/pkg/apis/camel/v1"
 	traitv1 "github.com/apache/camel-k/v2/pkg/apis/camel/v1/trait"
 	"github.com/apache/camel-k/v2/pkg/metadata"
+	"github.com/apache/camel-k/v2/pkg/util/knative"
 	"github.com/apache/camel-k/v2/pkg/util/kubernetes"
 )
 
@@ -146,6 +147,11 @@ func (t *knativeServiceTrait) Apply(e *Environment) error {
 func (t *knativeServiceTrait) SelectControllerStrategy(e *Environment) (*ControllerStrategy, error) {
 	if !pointer.BoolDeref(t.Enabled, true) || e.CamelCatalog == nil {
 		// explicitly disabled or sourceless Integration (missing catalog)
+		return nil, nil
+	}
+
+	// Knative serving is required
+	if ok, _ := knative.IsServingInstalled(e.Client); !ok {
 		return nil, nil
 	}
 

--- a/pkg/trait/knative_service_test.go
+++ b/pkg/trait/knative_service_test.go
@@ -414,6 +414,81 @@ func TestKnativeServiceNotApplicable(t *testing.T) {
 	}))
 }
 
+func TestKnativeServiceNoServingAvailable(t *testing.T) {
+	catalog, err := camel.DefaultCatalog()
+	require.NoError(t, err)
+
+	client, _ := test.NewFakeClient()
+	fakeClient := client.(*test.FakeClient) //nolint
+	fakeClient.DisableKnativeServing()
+
+	traitCatalog := NewCatalog(nil)
+
+	environment := Environment{
+		CamelCatalog: catalog,
+		Catalog:      traitCatalog,
+		Client:       client,
+		Integration: &v1.Integration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      KnativeServiceTestName,
+				Namespace: KnativeServiceTestNamespace,
+			},
+			Status: v1.IntegrationStatus{
+				Phase: v1.IntegrationPhaseDeploying,
+			},
+			Spec: v1.IntegrationSpec{
+				Profile: v1.TraitProfileKnative,
+				Sources: []v1.SourceSpec{
+					{
+						DataSpec: v1.DataSpec{
+							Name:    "routes.js",
+							Content: `from("platform-http:test").log("hello")`,
+						},
+						Language: v1.LanguageJavaScript,
+					},
+				},
+			},
+		},
+		IntegrationKit: &v1.IntegrationKit{
+			Status: v1.IntegrationKitStatus{
+				Phase: v1.IntegrationKitPhaseReady,
+			},
+		},
+		Platform: &v1.IntegrationPlatform{
+			Spec: v1.IntegrationPlatformSpec{
+				Cluster: v1.IntegrationPlatformClusterOpenShift,
+				Build: v1.IntegrationPlatformBuildSpec{
+					PublishStrategy: v1.IntegrationPlatformBuildPublishStrategyS2I,
+					Registry:        v1.RegistrySpec{Address: "registry"},
+					RuntimeVersion:  catalog.Runtime.Version,
+				},
+			},
+			Status: v1.IntegrationPlatformStatus{
+				Phase: v1.IntegrationPlatformPhaseReady,
+			},
+		},
+		EnvVars:        make([]corev1.EnvVar, 0),
+		ExecutedTraits: make([]Trait, 0),
+		Resources:      kubernetes.NewCollection(),
+	}
+	environment.Platform.ResyncStatusFullConfig()
+
+	// don't care about conditions in this unit test
+	_, err = traitCatalog.apply(&environment)
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, environment.ExecutedTraits)
+	assert.Nil(t, environment.GetTrait("knative-service"))
+
+	assert.Nil(t, environment.Resources.GetKnativeService(func(service *serving.Service) bool {
+		return service.Name == KnativeServiceTestName
+	}))
+
+	assert.NotNil(t, environment.Resources.GetDeployment(func(deployment *appsv1.Deployment) bool {
+		return deployment.Name == KnativeServiceTestName
+	}))
+}
+
 func TestKnativeServiceWithRollout(t *testing.T) {
 	environment := createKnativeServiceTestEnvironment(t, &traitv1.KnativeServiceTrait{RolloutDuration: "60s"})
 	assert.NotEmpty(t, environment.ExecutedTraits)

--- a/pkg/trait/knative_test.go
+++ b/pkg/trait/knative_test.go
@@ -147,6 +147,14 @@ func TestKnativeEnvConfigurationFromTrait(t *testing.T) {
 	eEventSink := ne.FindService("default", knativeapi.CamelEndpointKindSink, knativeapi.CamelServiceTypeEvent, "eventing.knative.dev/v1", "Broker")
 	assert.NotNil(t, eEventSink)
 	assert.Equal(t, "http://broker-default.host/", eEventSink.URL)
+
+	assert.NotNil(t, environment.Resources.GetKnativeSubscription(func(subscription *messaging.Subscription) bool {
+		return assert.Equal(t, "channel-source-1-test", subscription.Name)
+	}))
+
+	assert.NotNil(t, environment.Resources.GetKnativeTrigger(func(trigger *eventing.Trigger) bool {
+		return assert.Equal(t, "default-test", trigger.Name)
+	}))
 }
 
 func TestKnativeEnvConfigurationFromSource(t *testing.T) {
@@ -253,6 +261,203 @@ func TestKnativeEnvConfigurationFromSource(t *testing.T) {
 	broker := ne.FindService("evt.type", knativeapi.CamelEndpointKindSource, knativeapi.CamelServiceTypeEvent, "", "")
 	assert.NotNil(t, broker)
 	assert.Equal(t, "false", broker.Metadata[knativeapi.CamelMetaKnativeReply])
+
+	assert.NotNil(t, environment.Resources.GetKnativeSubscription(func(subscription *messaging.Subscription) bool {
+		return assert.Equal(t, "channel-source-1-test", subscription.Name)
+	}))
+
+	assert.NotNil(t, environment.Resources.GetKnativeTrigger(func(trigger *eventing.Trigger) bool {
+		return assert.Equal(t, "default-test-evttype", trigger.Name)
+	}))
+}
+
+func TestKnativeTriggerConfiguration(t *testing.T) {
+	catalog, err := camel.DefaultCatalog()
+	require.NoError(t, err)
+
+	c, err := NewFakeClient("ns")
+	require.NoError(t, err)
+
+	traitCatalog := NewCatalog(c)
+
+	environment := Environment{
+		CamelCatalog: catalog,
+		Catalog:      traitCatalog,
+		Client:       c,
+		Integration: &v1.Integration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "ns",
+			},
+			Status: v1.IntegrationStatus{
+				Phase: v1.IntegrationPhaseDeploying,
+			},
+			Spec: v1.IntegrationSpec{
+				Profile: v1.TraitProfileKnative,
+				Sources: []v1.SourceSpec{
+					{
+						DataSpec: v1.DataSpec{
+							Name: "route.java",
+							Content: `
+								public class CartoonMessagesMover extends RouteBuilder {
+									public void configure() {
+										from("knative:event/evt.type")
+ 											.log("${body}");
+									}
+								}
+							`,
+						},
+						Language: v1.LanguageJavaSource,
+					},
+				},
+				Traits: v1.Traits{
+					Knative: &traitv1.KnativeTrait{
+						Trait: traitv1.Trait{
+							Enabled: pointer.Bool(true),
+						},
+					},
+				},
+			},
+		},
+		IntegrationKit: &v1.IntegrationKit{
+			Status: v1.IntegrationKitStatus{
+				Phase: v1.IntegrationKitPhaseReady,
+			},
+		},
+		Platform: &v1.IntegrationPlatform{
+			Spec: v1.IntegrationPlatformSpec{
+				Cluster: v1.IntegrationPlatformClusterOpenShift,
+				Build: v1.IntegrationPlatformBuildSpec{
+					PublishStrategy: v1.IntegrationPlatformBuildPublishStrategyS2I,
+					Registry:        v1.RegistrySpec{Address: "registry"},
+					RuntimeVersion:  catalog.Runtime.Version,
+				},
+				Profile: v1.TraitProfileKnative,
+			},
+			Status: v1.IntegrationPlatformStatus{
+				Phase: v1.IntegrationPlatformPhaseReady,
+			},
+		},
+		EnvVars:        make([]corev1.EnvVar, 0),
+		ExecutedTraits: make([]Trait, 0),
+		Resources:      k8sutils.NewCollection(),
+	}
+	environment.Platform.ResyncStatusFullConfig()
+
+	// don't care about conditions in this unit test
+	_, err = traitCatalog.apply(&environment)
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, environment.ExecutedTraits)
+	assert.NotNil(t, environment.GetTrait("knative"))
+
+	assert.NotNil(t, environment.Resources.GetKnativeTrigger(func(trigger *eventing.Trigger) bool {
+		matching := true
+
+		matching = matching && assert.Equal(t, "default", trigger.Spec.Broker)
+		matching = matching && assert.Equal(t, serving.SchemeGroupVersion.String(), trigger.Spec.Subscriber.Ref.APIVersion)
+		matching = matching && assert.Equal(t, "Service", trigger.Spec.Subscriber.Ref.Kind)
+		matching = matching && assert.Equal(t, "/events/evt.type", trigger.Spec.Subscriber.URI.Path)
+		matching = matching && assert.Equal(t, "default-test-evttype", trigger.Name)
+
+		return matching
+	}))
+}
+
+func TestKnativeTriggerConfigurationNoServingAvailable(t *testing.T) {
+	catalog, err := camel.DefaultCatalog()
+	require.NoError(t, err)
+
+	c, err := NewFakeClient("ns")
+	require.NoError(t, err)
+
+	fakeClient := c.(*test.FakeClient) //nolint
+	fakeClient.DisableKnativeServing()
+
+	traitCatalog := NewCatalog(c)
+
+	environment := Environment{
+		CamelCatalog: catalog,
+		Catalog:      traitCatalog,
+		Client:       c,
+		Integration: &v1.Integration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "ns",
+			},
+			Status: v1.IntegrationStatus{
+				Phase: v1.IntegrationPhaseDeploying,
+			},
+			Spec: v1.IntegrationSpec{
+				Profile: v1.TraitProfileKnative,
+				Sources: []v1.SourceSpec{
+					{
+						DataSpec: v1.DataSpec{
+							Name: "route.java",
+							Content: `
+								public class CartoonMessagesMover extends RouteBuilder {
+									public void configure() {
+										from("knative:event/evt.type")
+ 											.log("${body}");
+									}
+								}
+							`,
+						},
+						Language: v1.LanguageJavaSource,
+					},
+				},
+				Traits: v1.Traits{
+					Knative: &traitv1.KnativeTrait{
+						Trait: traitv1.Trait{
+							Enabled: pointer.Bool(true),
+						},
+					},
+				},
+			},
+		},
+		IntegrationKit: &v1.IntegrationKit{
+			Status: v1.IntegrationKitStatus{
+				Phase: v1.IntegrationKitPhaseReady,
+			},
+		},
+		Platform: &v1.IntegrationPlatform{
+			Spec: v1.IntegrationPlatformSpec{
+				Cluster: v1.IntegrationPlatformClusterOpenShift,
+				Build: v1.IntegrationPlatformBuildSpec{
+					PublishStrategy: v1.IntegrationPlatformBuildPublishStrategyS2I,
+					Registry:        v1.RegistrySpec{Address: "registry"},
+					RuntimeVersion:  catalog.Runtime.Version,
+				},
+				Profile: v1.TraitProfileKnative,
+			},
+			Status: v1.IntegrationPlatformStatus{
+				Phase: v1.IntegrationPlatformPhaseReady,
+			},
+		},
+		EnvVars:        make([]corev1.EnvVar, 0),
+		ExecutedTraits: make([]Trait, 0),
+		Resources:      k8sutils.NewCollection(),
+	}
+	environment.Platform.ResyncStatusFullConfig()
+
+	// don't care about conditions in this unit test
+	_, err = traitCatalog.apply(&environment)
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, environment.ExecutedTraits)
+	assert.NotNil(t, environment.GetTrait("knative"))
+
+	assert.NotNil(t, environment.Resources.GetKnativeTrigger(func(trigger *eventing.Trigger) bool {
+		matching := true
+
+		matching = matching && assert.Equal(t, "default", trigger.Spec.Broker)
+		matching = matching && assert.Equal(t, "v1", trigger.Spec.Subscriber.Ref.APIVersion)
+		matching = matching && assert.Equal(t, "Service", trigger.Spec.Subscriber.Ref.Kind)
+		matching = matching && assert.Equal(t, "/events/evt.type", trigger.Spec.Subscriber.URI.Path)
+		matching = matching && assert.Equal(t, "default-test-evttype", trigger.Name)
+
+		return matching
+	}))
 }
 
 func TestKnativePlatformHttpConfig(t *testing.T) {

--- a/pkg/util/knative/enabled.go
+++ b/pkg/util/knative/enabled.go
@@ -28,11 +28,24 @@ import (
 
 // IsRefKindInstalled returns true if the cluster has the referenced Kind installed.
 func IsRefKindInstalled(c kubernetes.Interface, ref corev1.ObjectReference) (bool, error) {
-	if installed, err := isInstalled(c, ref.GroupVersionKind().GroupVersion()); err != nil {
+	if installed, err := isServerResourceAvailable(c, ref.GroupVersionKind().GroupVersion()); err != nil {
 		return false, err
 	} else if installed {
 		return true, nil
 	}
+	return false, nil
+}
+
+// IsInstalled returns true if we are connected to a cluster with either Knative Serving or Eventing installed.
+func IsInstalled(c kubernetes.Interface) (bool, error) {
+	if ok, err := IsServingInstalled(c); ok {
+		return ok, err
+	} else if ok, err = IsEventingInstalled(c); ok {
+		return ok, err
+	} else if err != nil {
+		return false, err
+	}
+
 	return false, nil
 }
 
@@ -52,7 +65,7 @@ func IsEventingInstalled(c kubernetes.Interface) (bool, error) {
 	})
 }
 
-func isInstalled(c kubernetes.Interface, api schema.GroupVersion) (bool, error) {
+func isServerResourceAvailable(c kubernetes.Interface, api schema.GroupVersion) (bool, error) {
 	_, err := c.Discovery().ServerResourcesForGroupVersion(api.String())
 	if err != nil && (k8serrors.IsNotFound(err) || util.IsUnknownAPIError(err)) {
 		return false, nil

--- a/pkg/util/knative/knative.go
+++ b/pkg/util/knative/knative.go
@@ -75,7 +75,27 @@ func CreateSubscription(channelReference corev1.ObjectReference, serviceName str
 	}
 }
 
-func CreateTrigger(brokerReference corev1.ObjectReference, serviceName string, eventType string, path string) *eventing.Trigger {
+// CreateServiceTrigger create Knative trigger with arbitrary Kubernetes Service as a subscriber - usually used when no Knative Serving is available on the cluster.
+func CreateServiceTrigger(brokerReference corev1.ObjectReference, serviceName string, eventType string, path string) (*eventing.Trigger, error) {
+	subscriberRef := duckv1.KReference{
+		APIVersion: "v1",
+		Kind:       "Service",
+		Name:       serviceName,
+	}
+	return CreateTrigger(brokerReference, subscriberRef, eventType, path)
+}
+
+// CreateKnativeServiceTrigger create Knative trigger with Knative Serving Service as a subscriber - default option when Knative Serving is available on the cluster.
+func CreateKnativeServiceTrigger(brokerReference corev1.ObjectReference, serviceName string, eventType string, path string) (*eventing.Trigger, error) {
+	subscriberRef := duckv1.KReference{
+		APIVersion: serving.SchemeGroupVersion.String(),
+		Kind:       "Service",
+		Name:       serviceName,
+	}
+	return CreateTrigger(brokerReference, subscriberRef, eventType, path)
+}
+
+func CreateTrigger(brokerReference corev1.ObjectReference, subscriberRef duckv1.KReference, eventType string, path string) (*eventing.Trigger, error) {
 	nameSuffix := ""
 	var attributes map[string]string
 	if eventType != "" {
@@ -91,7 +111,7 @@ func CreateTrigger(brokerReference corev1.ObjectReference, serviceName string, e
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: brokerReference.Namespace,
-			Name:      brokerReference.Name + "-" + serviceName + nameSuffix,
+			Name:      brokerReference.Name + "-" + subscriberRef.Name + nameSuffix,
 		},
 		Spec: eventing.TriggerSpec{
 			Filter: &eventing.TriggerFilter{
@@ -99,17 +119,13 @@ func CreateTrigger(brokerReference corev1.ObjectReference, serviceName string, e
 			},
 			Broker: brokerReference.Name,
 			Subscriber: duckv1.Destination{
-				Ref: &duckv1.KReference{
-					APIVersion: serving.SchemeGroupVersion.String(),
-					Kind:       "Service",
-					Name:       serviceName,
-				},
+				Ref: &subscriberRef,
 				URI: &apis.URL{
 					Path: path,
 				},
 			},
 		},
-	}
+	}, nil
 }
 
 func CreateSinkBinding(source corev1.ObjectReference, target corev1.ObjectReference) *sources.SinkBinding {

--- a/pkg/util/kubernetes/collection.go
+++ b/pkg/util/kubernetes/collection.go
@@ -23,6 +23,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	messaging "knative.dev/eventing/pkg/apis/messaging/v1"
 
 	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
 
@@ -266,10 +267,32 @@ func (c *Collection) GetServiceForIntegration(integration *v1.Integration) *core
 	})
 }
 
-// GetKnativeService returns a knative Service that matches the given function.
+// GetKnativeService returns a Knative Service that matches the given function.
 func (c *Collection) GetKnativeService(filter func(*serving.Service) bool) *serving.Service {
 	var retValue *serving.Service
 	c.VisitKnativeService(func(re *serving.Service) {
+		if filter(re) {
+			retValue = re
+		}
+	})
+	return retValue
+}
+
+// GetKnativeTrigger returns a Knative Trigger that matches the given function.
+func (c *Collection) GetKnativeTrigger(filter func(*eventing.Trigger) bool) *eventing.Trigger {
+	var retValue *eventing.Trigger
+	c.VisitKnativeTrigger(func(re *eventing.Trigger) {
+		if filter(re) {
+			retValue = re
+		}
+	})
+	return retValue
+}
+
+// GetKnativeSubscription returns a Knative channel Subscription that matches the given function.
+func (c *Collection) GetKnativeSubscription(filter func(subscription *messaging.Subscription) bool) *messaging.Subscription {
+	var retValue *messaging.Subscription
+	c.VisitKnativeSubscription(func(re *messaging.Subscription) {
 		if filter(re) {
 			retValue = re
 		}
@@ -352,6 +375,15 @@ func (c *Collection) VisitKnativeServiceE(visitor func(*serving.Service) error) 
 func (c *Collection) VisitKnativeTrigger(visitor func(trigger *eventing.Trigger)) {
 	c.Visit(func(res runtime.Object) {
 		if conv, ok := res.(*eventing.Trigger); ok {
+			visitor(conv)
+		}
+	})
+}
+
+// VisitKnativeSubscription executes the visitor function on all Knative channel Subscription resources.
+func (c *Collection) VisitKnativeSubscription(visitor func(trigger *messaging.Subscription)) {
+	c.Visit(func(res runtime.Object) {
+		if conv, ok := res.(*messaging.Subscription); ok {
 			visitor(conv)
 		}
 	})

--- a/pkg/util/test/client.go
+++ b/pkg/util/test/client.go
@@ -113,10 +113,12 @@ func NewFakeClient(initObjs ...runtime.Object) (client.Client, error) {
 	})
 
 	return &FakeClient{
-		Client:    c,
-		Interface: clientset,
-		camel:     camelClientset,
-		scales:    &fakescaleclient,
+		Client:                 c,
+		Interface:              clientset,
+		camel:                  camelClientset,
+		scales:                 &fakescaleclient,
+		enabledKnativeServing:  true,
+		enabledKnativeEventing: true,
 	}, nil
 }
 
@@ -138,10 +140,12 @@ func filterObjects(scheme *runtime.Scheme, input []runtime.Object, filter func(g
 type FakeClient struct {
 	controller.Client
 	kubernetes.Interface
-	camel            *fakecamelclientset.Clientset
-	scales           *fakescale.FakeScaleClient
-	disabledGroups   []string
-	enabledOpenshift bool
+	camel                  *fakecamelclientset.Clientset
+	scales                 *fakescale.FakeScaleClient
+	disabledGroups         []string
+	enabledOpenshift       bool
+	enabledKnativeServing  bool
+	enabledKnativeEventing bool
 }
 
 func (c *FakeClient) Intercept(intercept *interceptor.Funcs) {
@@ -191,6 +195,14 @@ func (c *FakeClient) EnableOpenshiftDiscovery() {
 	c.enabledOpenshift = true
 }
 
+func (c *FakeClient) DisableKnativeServing() {
+	c.enabledKnativeServing = false
+}
+
+func (c *FakeClient) DisableKnativeEventing() {
+	c.enabledKnativeEventing = false
+}
+
 func (c *FakeClient) AuthorizationV1() authorizationv1.AuthorizationV1Interface {
 	return &FakeAuthorization{
 		AuthorizationV1Interface: c.Interface.AuthorizationV1(),
@@ -201,9 +213,11 @@ func (c *FakeClient) AuthorizationV1() authorizationv1.AuthorizationV1Interface 
 
 func (c *FakeClient) Discovery() discovery.DiscoveryInterface {
 	return &FakeDiscovery{
-		DiscoveryInterface: c.Interface.Discovery(),
-		disabledGroups:     c.disabledGroups,
-		enabledOpenshift:   c.enabledOpenshift,
+		DiscoveryInterface:     c.Interface.Discovery(),
+		disabledGroups:         c.disabledGroups,
+		enabledOpenshift:       c.enabledOpenshift,
+		enabledKnativeServing:  c.enabledKnativeServing,
+		enabledKnativeEventing: c.enabledKnativeEventing,
 	}
 }
 
@@ -229,8 +243,10 @@ func (f *FakeAuthorization) SelfSubjectRulesReviews() authorizationv1.SelfSubjec
 
 type FakeDiscovery struct {
 	discovery.DiscoveryInterface
-	disabledGroups   []string
-	enabledOpenshift bool
+	disabledGroups         []string
+	enabledOpenshift       bool
+	enabledKnativeServing  bool
+	enabledKnativeEventing bool
 }
 
 func (f *FakeDiscovery) ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error) {
@@ -247,26 +263,33 @@ func (f *FakeDiscovery) ServerResourcesForGroupVersion(groupVersion string) (*me
 		}
 	}
 
-	// used to verify if knative is installed
-	if groupVersion == "serving.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
-		return &metav1.APIResourceList{
-			GroupVersion: "serving.knative.dev/v1",
-		}, nil
+	// used to verify if Knative Serving is installed
+	if f.enabledKnativeServing {
+		if groupVersion == "serving.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
+			return &metav1.APIResourceList{
+				GroupVersion: "serving.knative.dev/v1",
+			}, nil
+		}
 	}
-	if groupVersion == "eventing.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
-		return &metav1.APIResourceList{
-			GroupVersion: "eventing.knative.dev/v1",
-		}, nil
+
+	// used to verify if Knative Eventing is installed
+	if f.enabledKnativeEventing {
+		if groupVersion == "eventing.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
+			return &metav1.APIResourceList{
+				GroupVersion: "eventing.knative.dev/v1",
+			}, nil
+		}
+		if groupVersion == "messaging.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
+			return &metav1.APIResourceList{
+				GroupVersion: "messaging.knative.dev/v1",
+			}, nil
+		}
+		if groupVersion == "messaging.knative.dev/v1beta1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
+			return &metav1.APIResourceList{
+				GroupVersion: "messaging.knative.dev/v1beta1",
+			}, nil
+		}
 	}
-	if groupVersion == "messaging.knative.dev/v1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
-		return &metav1.APIResourceList{
-			GroupVersion: "messaging.knative.dev/v1",
-		}, nil
-	}
-	if groupVersion == "messaging.knative.dev/v1beta1" && !util.StringSliceExists(f.disabledGroups, groupVersion) {
-		return &metav1.APIResourceList{
-			GroupVersion: "messaging.knative.dev/v1beta1",
-		}, nil
-	}
+
 	return f.DiscoveryInterface.ServerResourcesForGroupVersion(groupVersion)
 }

